control-plane/config-entries/controllersv2/meshconfig_controller_test.go

```diff
@@ -0,0 +1,765 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllersv2
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	logrtest "github.com/go-logr/logr/testr"
+	"github.com/google/go-cmp/cmp"
+	pbauth "github.com/hashicorp/consul/proto-public/pbauth/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/testing/protocmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/hashicorp/consul-k8s/control-plane/api/common"
+	"github.com/hashicorp/consul-k8s/control-plane/api/v1alpha1"
+	"github.com/hashicorp/consul-k8s/control-plane/api/v2alpha1"
+	"github.com/hashicorp/consul-k8s/control-plane/connect-inject/constants"
+	"github.com/hashicorp/consul-k8s/control-plane/consul"
+	"github.com/hashicorp/consul-k8s/control-plane/helper/test"
+)
+
+type testReconciler interface {
+	Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error)
+}
+
+// TestMeshConfigController_createsMeshConfig validated resources are created in Consul from kube objects.
+func TestMeshConfigController_createsMeshConfig(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name       string
+		meshConfig common.MeshConfig
+		expected   *pbauth.TrafficPermissions
```

Comment:
I realize these tests are suppose to be generic, but in the interest of time, I just hardcoded the TrafficPermissions structs in several places. I suspect they will get refactored shortly at the expense of some boilerplate to convert to a `pbresource.Resource`.